src/shipit_code_coverage/shipit_code_coverage/report_generators.py

```diff
@@ -1,54 +1,180 @@
 # -*- coding: utf-8 -*-
 import json
 import os
+import signal
+import subprocess
+from datetime import datetime
 
+import pytz
+import requests
+
+from cli_common.log import get_logger
 from shipit_code_coverage import grcov
 
+logger = get_logger(__name__)
+
+
+class ZeroCov(object):
+
+    DATE_FORMAT = '%Y-%m-%d'
+
+    def __init__(self, repo_dir):
+        self.repo_dir = repo_dir
+
+    def get_pid_file(self):
+        return '/tmp/hgmo.pid'
+
+    def get_pid(self):
+        with open(self.get_pid_file(), 'r') as In:
+            pid = In.read()
+            return int(pid)
+
+    def get_file_size(self, filename):
+        if self.repo_dir:
+            filename = os.path.join(self.repo_dir, filename)
+            if os.path.isfile(filename):
+                return os.path.getsize(filename)
+        return 0
+
+    def get_utc_from_timestamp(self, ts):
+        d = datetime.utcfromtimestamp(ts)
+        return d.replace(tzinfo=pytz.utc)
+
+    def get_date_str(self, d):
+        return d.strftime(ZeroCov.DATE_FORMAT)
+
+    def kill_hgmo(self):
+        pid = self.get_pid()
+        os.killpg(os.getpgid(pid), signal.SIGTERM)
+
+    def run_hgmo(self):
+        if not os.path.isdir(self.repo_dir):
+            logger.warning('Not a directory for m-c', dir=self.repo_dir)
+            return False
+
+        proc = subprocess.Popen(['hg', 'serve',
+                                 '--hgmo',
+                                 '--daemon',
+                                 '--pid-file', self.get_pid_file()],
+                                cwd=self.repo_dir)
+        proc.wait()
+
+        pid = self.get_pid()
+        logger.info('hgmo is running', pid=pid)
+        return True
+
+    def get_pushlog(self):
+        has_hgmo = self.run_hgmo()
+        if has_hgmo:
+            url = 'http://localhost:8000/json-pushes'
+        else:
+            return {}
```

Comment:
...you can remove the if else here and always use `http://localhost:8000/json-pushes`